ts/main.ts

```diff
@@ -137,3 +177,38 @@ $entriesViewA.addEventListener('click', () => {
 $newEntryA.addEventListener('click', () => {
   viewSwap('entry-form');
 });
+
+$allEntriesUl.addEventListener('click', (event: Event) => {
+  const $target = event.target as HTMLElement;
+  if (!$target) throw new Error('$target query failed');
+
+  if (!$target.matches('.fa-pencil')) {
+    return;
+  }
+
+  viewSwap('entry-form');
+
```

Comment:
I would recommend moving your `viewSwap` lower in your function. Technically what's going on here is that you are navigating the user to the `entry-form` page and then you are updating the values on the page. Instead, you should wait until after you have updated the values on the page before you move the user to it.